tempodb/backend/s3/s3.go

```diff
@@ -128,11 +130,31 @@ func internalNew(cfg *Config, confirm bool) (*readerWriter, error) {
 }
 
 func getPutObjectOptions(rw *readerWriter) minio.PutObjectOptions {
+	sseConfig, err := buildSSEConfig(rw.cfg)
+	if sseConfig == nil && err == nil {
```

Comment:
it should not, since `buildSSEConfig` checks if the config is incorrect, did you happen to test it and see it working (i didn't have the change to test it myself yet nor write test)

Reply:
In that scenario, sseConfig = nil and err != nil, therefore it will go for the else branch. It doesn't matter since the sseConfig is nil so it won't affect. Maybe we could log a warning indicating that the SSE type is not valid